src/can/can.rs

```diff
@@ -0,0 +1,161 @@
+use super::enums::*;
+use super::CanFrame;
+use crate as hal;
+use crate::can::registers::Registers;
+use crate::can::util;
+use crate::{into_ref, pac, Peripheral, PeripheralRef, RccPeripheral};
+
+pub struct Can<'d, T: Instance> {
+    _peri: PeripheralRef<'d, T>,
+    fifo: CanFifo,
+    last_mailbox_used: usize,
+}
+
+#[derive(Debug)]
+pub enum CanInitError {
+    InvalidTimings,
+}
+
+impl<'d, T: Instance> Can<'d, T> {
+    /// Assumes AFIO & PORTB clocks have been enabled by HAL.
+    ///
+    /// CAN_RX is mapped to PB8, and CAN_TX is mapped to PB9.
```

Comment:
These comments can be removed.

Reply:
I will review all the relevance of the comments again)

Reply:
I will review the relevance of the comments)